cmd/chihaya/main.go

```diff
@@ -173,6 +173,14 @@ func main() {
 		Short: "BitTorrent Tracker",
 		Long:  "A customizable, multi-protocol BitTorrent Tracker",
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			noColors, err := cmd.Flags().GetBool("nocolors")
+			if err != nil {
+				return err
+			}
+			if noColors {
+				log.SetFormatter(&logrus.TextFormatter{DisableColors: true})
```

Comment:
When you replace the TextFormatter with this new instance, that's what's clearing the formatting.

Reply:
But it just uses an empty `TextFormatter` by default, see [here](https://github.com/sirupsen/logrus/blob/master/logger.go#L69-L76). I give it the same one, disable colors, and the whole format changes. But as far as I can see from the source, that's intended behaviour...

Reply:
Looks like this is just hardcoded: [Colored format](https://github.com/sirupsen/logrus/blob/d682213848ed68c0a260ca37d6dd5ace8423f5ba/text_formatter.go#L134), [Uncolored format](https://github.com/sirupsen/logrus/blob/d682213848ed68c0a260ca37d6dd5ace8423f5ba/text_formatter.go#L103)

Reply:
I'm lost, what should be fixed upstream?
I'm okay with the uncolored format it outputs, was just surprised it did that.

Reply:
Two fixes for logrus upstream could be proposed:

1. logrus internally has a function called [checkIfTerminal](https://github.com/sirupsen/logrus/blob/d682213848ed68c0a260ca37d6dd5ace8423f5ba/text_formatter.go#L64), it should probably have another function for detecting Windows and not printing terminal escape sequences when doing so.
2. the non-colored output should be the same as the colored output. If you look at the implementation, they literally just use two different format strings.

Reply:
Funny enough, they claim this to be [fixed](https://github.com/sirupsen/logrus/issues/172). I just checked, `TERM` is not set for me.

So, uh, is that blocking the PR? I can remove the messing with the logging for now, if that helps.

Reply:
Are you sure we just don't have an old version of logrus vendored?
I'm not a huge fan of the flag for colors, I'd rather leave that to be detected by logrus.